src/components/typeahead/typeahead.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from "react";
+import useOutsideClick from "../../shared/hooks/useOutsideClick";
+import TypeaheadItem from "./typeahead-item";
+import "./typeahead.css";
+
+const TYPEAHEAD_MAX_ITEMS = 100;
+
+interface TypeaheadProps {
+  labelKey: string;
+  descriptionKey?: string;
+  placeholderText?: string;
+  minimumLength?: number;
+  maxItems?: number;
+  className?: string;  
+  data: Array<any>;
+}
+
+function onTextChange(
+  text: string,
+  minimumLength: number,
+  setInputText: React.Dispatch<React.SetStateAction<string>>,
+  setActive: React.Dispatch<React.SetStateAction<boolean>>
+) {
+  setInputText(text);
+  if (text.length >= minimumLength) {
+    setActive(true);
+  } else {
+    setActive(false);
+  }
+}
+
+function Typeahead(props: TypeaheadProps) {
+  const {
+    placeholderText,
+    data,
+    minimumLength,
+    className,
+    maxItems,
+    labelKey,
+    descriptionKey
+  } = props;
+
+  const [inputText, setInputText] = useState("");
+  const [active, setActive] = useState(false);
+  let inputTextRef: HTMLInputElement | null;
+
+  useEffect(() => {
+    if (inputTextRef) {
+      inputTextRef.focus();
+    }
+  });
+  useOutsideClick(".select", () => setActive(false))
```

Comment:
Will this be ok if there are multiple select elements on the page ?

Reply:
This was initially a reference to the DOM element, but for some reason I wasn't able to make it work. This is temporary fix for the meantime.